core/feed_protocol/serializer/opds2.py

```diff
@@ -0,0 +1,69 @@
+import json
+from typing import Any, Dict
+
+from core.feed_protocol.types import FeedData, Link, WorkEntryData
+
+
+class OPDS2Serializer:
+    def __init__(self) -> None:
+        pass
+
+    def serialize_feed(self, feed: FeedData) -> bytes:
+        serialized: Dict[str, Any] = {"publications": []}
+        for entry in feed.entries:
+            if entry.computed:
+                publication = self._serialize_work_entry(entry.computed)
+                serialized["publications"].append(publication)
+
+        return json.dumps(serialized, indent=2).encode()
+
+    def _serialize_work_entry(self, data: WorkEntryData) -> Dict[str, Any]:
+        metadata: Dict[str, Any] = {}
+        if data.title:
+            metadata["title"] = data.title.text
+            metadata["sortAs"] = data.title.text  # TODO: Change this!
+
+        if data.subtitle:
+            metadata["subtitle"] = data.subtitle.text
+        if data.identifier:
+            metadata["identifier"] = data.identifier
+        if data.language:
+            metadata["language"] = data.language.text
+        if data.updated:
+            metadata["modified"] = data.updated.text
+        if data.published:
+            metadata["published"] = data.published.text
+        if data.summary:
+            metadata["description"] = data.summary.text
+
+        if data.publisher:
+            metadata["publisher"] = dict(name=data.publisher.text)
+        if data.imprint:
+            metadata["imprint"] = dict(name=data.imprint.text)
+
+        subjects = []
+        if data.categories:
+            for subject in data.categories:
+                subjects.append(
+                    {
+                        "scheme": subject.scheme,  # type: ignore[attr-defined]
+                        "name": subject.label,  # type: ignore[attr-defined]
+                        "sortAs": subject.label,  # type: ignore[attr-defined] # TODO: Change this!
```

Comment:
And this

Reply:
Hmm... I'm not sure there exists an alternative to the label, I think this just needs to stay as-is

Reply:
Comment has been updated in #1340